.github/workflows/ci.yaml

```diff
@@ -0,0 +1,30 @@
+name: CI build
+on:
+  pull_request:
+    branches:
+      - main
+  push:
+    branches:
+jobs:
+  ci:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Setup Go
```

Comment:
I'm a fan of just leaving these names out and using the "uses" to default the name. It's a bit cleaner and more straight forward than aliasing all of the steps in my mind. (e.g. Setup Go -> actions/setup-go@v2)